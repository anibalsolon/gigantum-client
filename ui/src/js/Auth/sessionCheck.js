// vendor
import { graphql } from 'react-relay';
// mutations
import RemoveUserIdentityMutation from 'Mutations/RemoveUserIdentityMutation';
// environment
import { fetchQuery } from 'JS/createRelayEnvironment';

const sessionCheckQuery = graphql`
  query sessionCheckQuery{
    userIdentity{
      isSessionValid
    }
  }
`;

const UserIdentity = {
  /*
    fetches session information from useridentity field in api
  */
  getUserIdentity: () => new Promise((resolve, reject) => {
    const fetchData = function () {
      fetchQuery(sessionCheckQuery(), {}).then((response, error) => {
        if (response) {
          resolve(response);
        } else {
          reject(response);
        }
      }).catch((error) => {
        reject(error);
      });
    };

    fetchData();
  }),
};

export default UserIdentity;
