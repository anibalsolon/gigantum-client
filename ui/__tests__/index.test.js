import React from 'react';
import Index from 'JS/index';
import renderer from 'react-test-renderer';
import Auth from 'JS/Auth/Auth';


describe('Test  index rendering', () => {
  let root = document.createElement('div');
  root.id = 'root';
  document.body.appendChild(root);

  it('renders without crashing', () => {
  expect(JSON.stringify(
      Object.assign({}, Index, { _reactInternalInstance: 'censored' }),
    )).toMatchSnapshot();
  });
  console.log(document.getElementById('root'));
});
