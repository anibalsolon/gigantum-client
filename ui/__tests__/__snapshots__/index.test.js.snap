// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test  index rendering renders without crashing 1`] = `"{\\"_reactInternalInstance\\":\\"censored\\"}"`;
